src/k8s/pkg/k8sd/features/cilium/network.go

```diff
@@ -37,22 +37,11 @@ func checkAndSanitizeCiliumVXLAN(port int) error {
 	}
 
 	for _, vxlanDevice := range vxlanDevices {
-		if vxlanDevice.Name == ciliumVXLANDeviceName {
-			// Note(Reza): Currently Cilium tries to bring up the vxlan interface before applying
-			// any configuration changes. If the Cilium vxlan interface has any conflicts with other
-			// interfaces that makes it unable to brought up, Cilium fails to apply configuration
-			// changes. Removing the interface before applying the new manifests is a temporary
-			// workaround. We can remove this block when the following issue gets settled:
-			// https://github.com/cilium/cilium/issues/38581
-			if err := utils.RemoveLink(vxlanDevice.Name); err != nil {
-				return err
-			}
-
-			continue
+		if vxlanDevice.Port == port && vxlanDevice.Name != ciliumVXLANDeviceName {
+			return fmt.Errorf("interface %s uses the same destination port as cilium. Please consider removing that device manually", vxlanDevice.Name)
```

Comment:
I think it's better to suggest changing the cilium port rather than removing an interface not controlled by k8s. That interface may be integral to the funtion of their environment. 
```suggestion
			return fmt.Errorf("interface %s uses the same destination port as cilium. Please consider changing the Cilium tunnel port", vxlanDevice.Name)
```

Reply:
+1 on this